### ISSUE TITLE
wp-cli installing plugins that were not requested

### ISSUE THREAD

**Issue by jschank** [wp-cli/wp-cli#2708]
I use vagrant to provision my wordpress virtual machine. And I loop over an array of plugin names to install by default.

I install them like so: (yes these are commented out - because of the random plugins that appear)

```
# PLUGIN_LIST=(
#   'akismet'
#   'audit-trail'
     'iwp-client'
#   'limit-login-attempts'
#   'root-relative-urls'
#   'user-role-editor'
#   'wordpress-seo'
#   'wp-filemanager'
# )

echo "~~ entering $site_path for installation"
cd "$site_path" || exit 1

for plugin in "${PLUGIN_LIST[@]}"
do
  echo "~~~ Installing $plugin $force"
  wp plugin install "$plugin" --allow-root "$force"
done 

```

When I run this script I get this output

```
=> capitol-connect-dev: ~ Installing default plugins
==> capitol-connect-dev: !!! Disabled automatic plugin provisioning
==> capitol-connect-dev: ~~ entering /var/www/html/apps/cc-qa for installation
==> capitol-connect-dev: ~~~ Installing iwp-client
==> capitol-connect-dev: Installing InfiniteWP Client (1.5.1.1)
==> capitol-connect-dev: Downloading install package from https://downloads.wordpress.org/plugin/iwp-client.zip...
==> capitol-connect-dev: Using cached file '/root/.wp-cli/cache/plugin/iwp-client-1.5.1.1.zip'...
==> capitol-connect-dev: Unpacking the package...
==> capitol-connect-dev: Installing the plugin...
==> capitol-connect-dev: Plugin installed successfully.
==> capitol-connect-dev: Installing Super Emoji Plus+ (0.1)
==> capitol-connect-dev: Downloading install package from https://downloads.wordpress.org/plugin/super-emoji-plus.zip...
==> capitol-connect-dev: Using cached file '/root/.wp-cli/cache/plugin/super-emoji-plus-0.1.zip'...
==> capitol-connect-dev: Unpacking the package...
==> capitol-connect-dev: Installing the plugin...
==> capitol-connect-dev: Plugin installed successfully.

```

**I never asked for Super Emoji Plus+**

When I allow all my default plugins I get an extra plugin on each install. The extra is random, But there are repetitions.

This is how I install wp-cli (in my provisioning steps)

```
curl -s -S -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
chmod +x wp-cli.phar
mv wp-cli.phar /usr/local/bin/wp

```

The version of wp-cli, which I am getting is:
```
[vagrant@localhost ~]$ wp --version
WP-CLI 0.23.0
```

This version definitely exhibits this problem. As does this
```
[vagrant@localhost ~]$ wp --info
PHP binary:	/usr/bin/php
PHP version:	5.5.34
php.ini used:	/etc/php.ini
WP-CLI root dir:	phar://wp-cli.phar
WP-CLI packages dir:	/home/vagrant/.wp-cli/packages/
WP-CLI global config:
WP-CLI project config:
WP-CLI version:	0.23.1
```

What is going on here? 
How do we make this stop?

**Comment by trepmal** [comment]
Any chance the `$plugin` variable is sometimes empty? Alternatively, what happens when you unquote `$plugin` in the install line?

When I run `wp plugin install ""` (need the quoted empty string), I end up installing the first result of 
`wp plugin search ""` which not long ago had super-emoji-plus.

**Comment by danielbachhuber** [comment]
> When I run wp plugin install "" (need the quoted empty string), I end up installing the first result of 
> wp plugin search "" which not long ago had super-emoji-plus.

This seems problematic :) Do you think we should fix it?

**Comment by jschank** [comment]
$Plugin can't be empty, because the script echos the value, and there is no output to indicate an empty value is being used. I'll try some test cases now that i know where the problem is. Perhaps a clue will appear. I can say this is relatively new behavior. Perhaps last week or two.

**Comment by trepmal** [comment]
@jschank I should have made my comment edit a reply so you'd get the email notification...

> Edit: Looks like $force may not be set? in which case you'd be passing an empty quoted string to the install command resulting in what I describe above.

**Comment by jschank** [comment]
Ah! That is definitely likely

**Comment by jschank** [comment]
Confirmed. If you run wp plugin install "<someplugin>" --allow-root ""
Then you get two plugins installed. The one you want, and a super-bonus random plugin you never asked for.

I have patched my script to no longer pass an empty string. But regardless of that. I believe that wp-cli has incorrect behavior. It should not install some random, unknown plugin, regardless of how it gets called.

Thanks to @trepmal, for noticing the empty string.
